File: fase8-kvm/glkvm-cloud/internal/http/dto/notification.go
package dto

// ─── SMTP Config ────────────────────────────────────────────────

type SMTPConfigReq struct {
	Host       string `json:"host"`
	Port       int    `json:"port"`
	Username   string `json:"username"`
	Password   string `json:"password"`
	FromEmail  string `json:"fromEmail"`
	Encryption string `json:"encryption"`
	Enabled    bool   `json:"enabled"`
}

type SMTPConfigResp struct {
	Host       string `json:"host"`
	Port       int    `json:"port"`
	Username   string `json:"username"`
	Password   string `json:"password"`
	FromEmail  string `json:"fromEmail"`
	Encryption string `json:"encryption"`
	Enabled    bool   `json:"enabled"`
	UpdatedAt  int64  `json:"updatedAt"`
}

// ─── SMTP Test ──────────────────────────────────────────────────

type SMTPTestReq struct {
	Email string `json:"email" binding:"required"`
}

// ─── Notify Rules ───────────────────────────────────────────────

type NotifyRulesReq struct {
	DeviceOnline  bool `json:"deviceOnline"`
	DeviceOffline bool `json:"deviceOffline"`
	RemoteAccess  bool `json:"remoteAccess"`
}

type NotifyRulesResp struct {
	DeviceOnline  bool  `json:"deviceOnline"`
	DeviceOffline bool  `json:"deviceOffline"`
	RemoteAccess  bool  `json:"remoteAccess"`
	UpdatedAt     int64 `json:"updatedAt"`
}

// ─── Recipients ─────────────────────────────────────────────────

type AddRecipientReq struct {
	Email string `json:"email" binding:"required"`
}

type RecipientResp struct {
	ID        int64  `json:"id"`
	Email     string `json:"email"`
	CreatedAt int64  `json:"createdAt"`
}

type ListRecipientsResp struct {
	Items []RecipientResp `json:"items"`
}

File: fase8-kvm/glkvm-cloud/ui/src/api/notification.ts
/**
 * Notification Settings API
 */

import request from './request'

// ─── SMTP Config ────────────────────────────────────────────────

export interface SMTPConfig {
    host: string
    port: number
    username: string
    password: string
    fromEmail: string
    encryption: string
    enabled: boolean
    updatedAt: number
}

export function reqGetSMTPConfig () {
    return request<SMTPConfig>({ url: '/api/notification/smtp' })
}

export function reqSaveSMTPConfig (data: Omit<SMTPConfig, 'updatedAt'>) {
    return request<SMTPConfig>({ url: '/api/notification/smtp', method: 'PUT', data })
}

export function reqTestSMTP (email: string) {
    return request<{ message: string }>({ url: '/api/notification/smtp/test', method: 'POST', data: { email } })
}

// ─── Notify Rules ───────────────────────────────────────────────

export interface NotifyRules {
    deviceOnline: boolean
    deviceOffline: boolean
    remoteAccess: boolean
    updatedAt: number
}

export function reqGetNotifyRules () {
    return request<NotifyRules>({ url: '/api/notification/rules' })
}

export function reqSaveNotifyRules (data: Omit<NotifyRules, 'updatedAt'>) {
    return request<NotifyRules>({ url: '/api/notification/rules', method: 'PUT', data })
}

// ─── Recipients ─────────────────────────────────────────────────

export interface Recipient {
    id: number
    email: string
    createdAt: number
}

export interface ListRecipientsResp {
    items: Recipient[]
}

export function reqListRecipients () {
    return request<ListRecipientsResp>({ url: '/api/notification/recipients' })
}

export function reqAddRecipient (email: string) {
    return request<Recipient>({ url: '/api/notification/recipients', method: 'POST', data: { email } })
}

export function reqRemoveRecipient (id: number) {
    return request<{ message: string }>({ url: `/api/notification/recipients/${id}`, method: 'DELETE' })
}
